service/worker/workerdeployment/drainage_workflow.go

```diff
@@ -62,16 +62,22 @@ func DrainageWorkflow(
 
 	// Set status = DRAINING and then sleep for visibilityGracePeriod (to let recently-started workflows arrive in visibility)
 	if !args.IsCan { // skip if resuming after the parent continued-as-new
-		parentWf := workflow.GetInfo(ctx).ParentWorkflowExecution
-		now := timestamppb.Now()
-		drainingInfo := &deploymentpb.VersionDrainageInfo{
-			Status:          enumspb.VERSION_DRAINAGE_STATUS_DRAINING,
-			LastChangedTime: now,
-			LastCheckedTime: now,
-		}
-		err := workflow.SignalExternalWorkflow(ctx, parentWf.ID, parentWf.RunID, SyncDrainageSignalName, drainingInfo).Get(ctx, nil)
-		if err != nil {
-			return err
+		// When workflow.GetVersion() is run for the new Workflow Execution, it records a marker in the
+		// Workflow history so that all future calls to GetVersion for this change Id—Step 1 in the example—
+		// on this Workflow Execution will always return the given version number, which is 1 in the example.
+		v := workflow.GetVersion(ctx, "Step1", workflow.DefaultVersion, 1)
+		if v == workflow.DefaultVersion { // needs patching because we removed a Signal call
+			parentWf := workflow.GetInfo(ctx).ParentWorkflowExecution
+			now := timestamppb.Now()
```

Comment:
is this indirectly using time.Now()? this needs to use workflow.Now instead (I think even the old version here should be changed)

Reply:
great call out! fixed in previous PR